lua/typescript-tools/api.lua

```diff
@@ -212,7 +212,9 @@ end
 ---
 ---@param codes integer[]
 function M.filter_diagnostics(codes)
-  vim.tbl_add_reverse_lookup(codes)
+  for k, v in pairs(codes) do
+    codes[v] = k
+  end
```

Comment:
Iterating over a table while adding keys to it is technically undefined - the implementation for `vim.tbl_add_reverse_lookup` uses `vim.tbl_keys` to get the full list of keys before iterating in order to avoid the issue.

This applies to the other two places where `vim.tbl_add_reverse_lookup` was used as well.